src/unix.rs

```diff
@@ -52,42 +58,51 @@ pub fn data_dirs() -> Vec<String> {
 /// populated.
 /// This ioctl is described here: <http://man7.org/linux/man-pages/man4/tty_ioctl.4.html>
 pub fn get_window_size() -> Result<(usize, usize), Error> {
-    nix::ioctl_read_bad!(get_ws, TIOCGWINSZ, Winsize);
-
-    let mut maybe_ws = std::mem::MaybeUninit::<Winsize>::uninit();
-
-    unsafe { get_ws(STDOUT_FILENO, maybe_ws.as_mut_ptr()).ok().map(|_| maybe_ws.assume_init()) }
+    let mut maybe_ws = std::mem::MaybeUninit::<winsize>::uninit();
+    cerr(unsafe { libc::ioctl(STDOUT_FILENO, TIOCGWINSZ, maybe_ws.as_mut_ptr()) })
+        .ok()
+        .map(|_| unsafe { maybe_ws.assume_init() })
         .filter(|ws| ws.ws_col != 0 && ws.ws_row != 0)
         .map_or(Err(Error::InvalidWindowSize), |ws| Ok((ws.ws_row as usize, ws.ws_col as usize)))
 }
 
-/// Return a MPSC receiver that receives a message whenever the window size is updated.
-pub fn get_window_size_update_receiver() -> Result<Option<Receiver<()>>, Error> {
-    // Create a channel for receiving window size update requests
-    let (ws_changed_tx, ws_changed_rx) = mpsc::sync_channel(1);
-    // Spawn a new thread that will push to the aforementioned channel every time the SIGWINCH
-    // signal is received
-    let signals = Signals::new(&[SIGWINCH])?;
-    std::thread::spawn(move || signals.forever().for_each(|_| ws_changed_tx.send(()).unwrap()));
-    Ok(Some(ws_changed_rx))
+pub fn register_winsize_change_signal_handler() -> Result<(), Error> {
+    unsafe {
+        let mut maybe_sa = std::mem::MaybeUninit::<sigaction>::uninit();
+        libc::sigemptyset(&mut (*maybe_sa.as_mut_ptr()).sa_mask);
+        (*maybe_sa.as_mut_ptr()).sa_flags = 0;
+        (*maybe_sa.as_mut_ptr()).sa_sigaction = set_windows_size_was_changed as sighandler_t;
+        cerr(libc::sigaction(libc::SIGWINCH, maybe_sa.as_mut_ptr(), std::ptr::null_mut()))
+    }
 }
 
+static WIN_CHANGED: AtomicBool = AtomicBool::new(false);
+
+#[no_mangle]
+extern "C" fn set_windows_size_was_changed(_: c_int, _: *mut siginfo_t, _: *mut c_void) {
+    WIN_CHANGED.store(true, Ordering::Relaxed);
+}
+
+pub fn get_windows_size_was_changed() -> bool { WIN_CHANGED.swap(false, Ordering::Relaxed) }
+
```

Comment:
I usually try to avoid the `set_` and `get_` prefixes in Rust ([RFC 344](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#getter/setter-methods-[rfc-344])), especially since `get_windows_size_was_changed()` has a side-effect here. What about `handle_window_size_change` and `has_window_size_changed`?

Reply:
Done.